hydromt_sfincs/sfincs.py

```diff
@@ -1572,6 +1573,130 @@ def setup_drainage_structures(
+            if merge and "vol" in self.grid:
+                da_vol = self.grid["vol"]
+            else:
+                da_vol = xr.full_like(self.mask, 0, dtype=np.float64)
+
+            # loop over the gdf rows and rasterize each geometry
+            for i, row in gdf.iterrows():
+                # create a gdf with only the current row
+                single_gdf = gpd.GeoDataFrame(gdf.loc[[i]]).reset_index(drop=True)
+
+                single_vol = single_gdf.get("volume")
+                single_height = single_gdf.get("height")
+
+                # check if volume or height is provided in the gdf or as input
+                if (
+                    single_vol is None or np.isnan(single_vol).any()
+                ):  # volume not provided or nan
+                    if (
+                        single_height is None or np.isnan(single_height).any()
+                    ):  # height not provided or nan
+                        if volume is not None:  # volume provided as input (list)
+                            single_vol = (
+                                volume if not isinstance(volume, list) else volume[i]
+                            )
+                        elif height is not None:  # height provided as input (list)
+                            single_height = (
+                                height if not isinstance(height, list) else height[i]
+                            )
+                        else:  # no volume or height provided
+                            self.logger.warning(
+                                f"No volume or height provided for storage location {i}"
+                            )
+                            continue
+                    else:  # height provided in gdf
+                        single_height = single_height[0]
+                else:  # volume provided in gdf
+                    single_vol = single_vol[0]
+
+                # check if gdf has point or polyhon geometry
+                if single_gdf.geometry.type[0] == "Point":
+                    # get x and y coordinate of the point in crs of the grid
+                    x, y = single_gdf.geometry.iloc[0].coords[0]
+
+                    # check if the grid is rotated
+                    if da_vol.raster.rotation != 0:
+                        # rotate the point
+                        x, y = ~da_vol.raster.transform * (x, y)
+                        # select the grid cell nearest to the point
+                        closest_point = da_vol.reindex(x=da_vol.x, y=da_vol.y).sel(
+                            x=x, y=y, method="nearest"
+                        )
+                    else:
+                        # select the grid cell nearest to the point
+                        closest_point = da_vol.sel(x=x, y=y, method="nearest")
+
+                    # add the volume to the grid cell
+                    if single_vol is not None and not np.isnan(single_vol).any():
```

Comment:
strictly speaking I don't think the any() is required here because single_vol is always a float right? Idem in a couple of instances below and for single_height.